src/Microsoft.AspNetCore.Sockets.Abstractions/IConnection.cs

```diff
@@ -12,11 +12,12 @@ public interface IConnection
     {
         Task StartAsync();
         Task SendAsync(byte[] data, CancellationToken cancellationToken);
+        Task StopAsync();
         Task DisposeAsync();
 
         IDisposable OnReceived(Func<byte[], object, Task> callback, object state);
 
-        Task Closed { get; }
+        event Action<Exception> Closed;
```

Comment:
Used to be `Func<Exception, Task>`, any reason we don't go back to that?

Reply:
I am not sure we want it (and what we had was not implemented correctly). I think we need to try which pattern works better for restarting the connection from this event.